src/System.CommandLine/Binding/ArgumentConverter.cs

```diff
@@ -310,21 +311,22 @@ public static bool CanBeBoundFromScalarValue(this Type type)
             }
         }
 
-        internal static object GetValueOrDefault(this ArgumentConversionResult result) =>
-            result.GetValueOrDefault<object>();
+        internal static object? GetValueOrDefault(this ArgumentConversionResult result) =>
+            result.GetValueOrDefault<object?>();
 
+        [return: MaybeNull]
         internal static T GetValueOrDefault<T>(this ArgumentConversionResult result)
         {
             switch (result)
             {
                 case SuccessfulArgumentConversionResult successful:
-                    return (T)successful.Value;
+                    return (T)successful.Value!;
```

Comment:
This should also no longer be needed in newer version of Roslyn (3.6, I think). May be worth a follow-up issue?